File: infiniAngular-main/User/src/model/Transaction.ts
import { BankAccount } from './BankAccount';


export class Transaction {
    transactionId: number;
    amount?: number;
    transactionDate?: Date;
    destination?: number;
    sourceRIB?: BankAccount;

    constructor() {
        this.transactionId = 0;
        this.amount = 0;
        this.transactionDate = new Date();
        this.destination = 0;
        this.sourceRIB = new BankAccount();
    }
}

File: MFB/src/main/java/tn/esprit/mfb/entity/Transaction.java
package tn.esprit.mfb.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;


@Entity
@Getter
@Setter
public class Transaction {

    @Id
    @Column(nullable = false, updatable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long transactionId;

    @Column
    private LocalDate transactionDate;

    @Column
    private Double amount;

    @Column
    private Long destination;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sourceribid")
    private BankAccount sourceRIB;

}
